### ISSUE TITLE
Real FFTs with empty `axes` should fail

### ISSUE THREAD

**Issue by toslunar** [cupy/cupy#3240]
The shape of the output of `fft.(i)rfft*` depends on the last axis.
```
>>> numpy.fft.rfftn(numpy.arange(3), axes=())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 5, in rfftn
  File "/path/to/python3.8/site-packages/numpy-1.18.2-py3.8-linux-x86_64.egg/numpy/fft/_pocketfft.py", line 1124, in rfftn
    a = rfft(a, s[-1], axes[-1], norm)
IndexError: list index out of range
>>> cupy.fft.rfftn(cupy.arange(3), axes=())
array([0, 1, 2])
```

**Comment by kmaehashi** [comment]
Is this issue introduced in recent NumPy changes? Otherwise I guess it should it be considered as a bug  (cc/ #3241)

**Comment by toslunar** [comment]
> Is this issue introduced in recent NumPy changes?

No.  The result with NumPy 1.15.4 is the same.

**Comment by leofang** [comment]
Should I fix this in #3102? I am touching rfft/irfft there.

**Comment by toslunar** [comment]
> Should I fix this in #3102? I am touching rfft/irfft there.

Sounds good, ... only if #3102 already fixed it.

- For the `_fft` case, I suppose the fix will be independent.
- For the `_fftn` case, the addition of `value_type='C2C'` option to `_prep_fftn_axes` in #3102 looks good also in this sense.  However, in the current implementation, `_fft` is returned by `_default_fft_func` if < 2 dim.

**Comment by leofang** [comment]
@toslunar @grlee77 I am tracing back the cause of this issue, and I don't think https://github.com/cupy/cupy/pull/2746#issuecomment-563439184 is correct...Can't reproduce the "new behavior" even with NumPy 1.18.1. 

I think this is best addressed in the functions `_fft()` and `_fftn()`. Will fix the latter in #3102, and leave the former in a separate PR as @toslunar suggested.

**Comment by leofang** [comment]
Sorry, changed my mind. It's simpler to fix this in a standalone PR. See #3264.
